lib/hammer_cli/defaults_commands.rb

```diff
@@ -0,0 +1,163 @@
+
+      def execute
+        data = context[:defaults].providers.map do |key, val|
+          {
+            :provider => key.to_s,
+            :defaults => (val.supported_defaults || ['*']).map(&:to_s)
+          }
+        end
+
+        fields = HammerCLI::Output::Dsl.new.build do
+          field :provider, _('Provider')
+          field :defaults, _('Supported defaults'), Fields::List
+        end
+
+        definition = HammerCLI::Output::Definition.new
+        definition.append(fields)
+
+        print_collection(definition, data)
+        HammerCLI::EX_OK
+      end
+
+      def adapter
+        @context[:adapter] || :table
+      end
+    end
+
+    class ListDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'list'
+      desc _('List all the default parameters')
+
+      def execute
+        data = context[:defaults].defaults_settings.map do |key, val|
+          {
+            :parameter => key.to_s,
+            :value     => val[:provider] ? "Provided by: " + val[:provider].to_s.capitalize : val[:value]
+          }
+        end
+
+        fields = HammerCLI::Output::Dsl.new.build do
+          field :parameter, _('Parameter')
+          field :value, _('Value'), Fields::List
+        end
+
+        definition = HammerCLI::Output::Definition.new
+        definition.append(fields)
+
+        print_collection(definition, data)
+        HammerCLI::EX_OK
+      end
+
+      def adapter
+        @context[:adapter] || :table
+      end
+    end
+
+    class DeleteDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'delete'
+
+      desc _('Delete a default param')
+      option "--param-name", "OPTION_NAME", _("The name of the default option"), :required => true
+
+      def execute
+        if context[:defaults].defaults_settings && context[:defaults].defaults_settings[option_param_name.to_sym]
+          context[:defaults].delete_default_from_conf(option_param_name.to_sym)
+          param_deleted(option_param_name)
+        else
+          variable_not_found
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    class AddDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'add'
+
+      desc _('Add a default parameter to config')
+      option "--param-name", "OPTION_NAME", _("The name of the default option (e.g. organization_id)."), :required => true
+      option "--param-value", "OPTION_VALUE", _("The value for the default option")
+      option "--plugin-name", "OPTION_PLUGIN_NAME", _("The name of the provider providing the value. For list available providers see `hammer defaults list --providers=true`.")
+
+      def execute
+        if option_plugin_name.nil? && option_param_value.nil? || !option_plugin_name.nil? && !option_param_value.nil?
+          bad_input
+          HammerCLI::EX_USAGE
+        else
+          if option_plugin_name
+            namespace = option_plugin_name
+            if !context[:defaults].providers.key?(namespace)
+              plugin_prob_message
```

Comment:
print_message(_("Provider #{namespace} was not found. See `hammer defaults providers` for available providers."))